azure-pipelines.yml

```diff
@@ -2,72 +2,101 @@ trigger:
+      inputs:
+        gitHubConnection: 'github.com_marianan'
+        repositoryName: '$(Build.Repository.Name)'
+        action: 'create'
+        target: '$(Build.SourceVersion)'
+        tagSource: 'userSpecifiedTag'
+        tag: '$(Build.SourceBranch)'
+        releaseNotesSource: 'inline'
+        assets: '$(Build.ArtifactStagingDirectory)/Package/*'
+        isPreRelease: true
+        addChangeLog: false
 
-  - task: CopyFiles@2
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)'
-      Contents: '*.tgz'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)/Package'
-  - task: PublishBuildArtifacts@1
-    inputs:
-      PathtoPublish: '$(Build.ArtifactStagingDirectory)/Package'
-      ArtifactName: 'drop'
-      publishLocation: 'Container'
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          if [[ $(Build.SourceBranch) =~ ^refs/tags/v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+          echo "##vso[task.setvariable variable=NPM_PUBLISH_FLAG;]true"
+          fi
 
-  - task: Npm@1
-    displayName: "npm Test"
-    inputs:
-      command: 'custom'
-      customCommand: 'test'
-    
-  - task: GitHubRelease@1
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
-    inputs:
-      gitHubConnection: 'github.com_marianan'
-      repositoryName: '$(Build.Repository.Name)'
-      action: 'create'
-      target: '$(Build.SourceVersion)'
-      tagSource: 'userSpecifiedTag'
-      tag: '$(Build.SourceBranch)'
-      releaseNotesSource: 'inline'
-      assets: '$(Build.ArtifactStagingDirectory)/Package/*'
-      addChangeLog: false
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          VERSION=$(node -p -e "require('./package.json').version")
+          echo "##vso[task.setvariable variable=PACKAGEVERSION;]$VERSION"
+
+    - task: Bash@3
+      inputs:
+        targetType: 'inline'
+        script: |
+          wget https://aka.ms/downloadazcopy-v10-linux
+          tar -xvf downloadazcopy-v10-linux
+          sudo cp ./azcopy_linux_amd64_*/azcopy /usr/bin/
 
-  - task: Npm@1
-    displayName: "npm Publishing"
-    condition: and(succeeded(), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
-    inputs:
-      command: 'publish'
-      publishRegistry: 'useFeed'
-      publishFeed: '"Service connection name"'      
\ No newline at end of file
+    - task: Bash@3
+      displayName: 'Uplaod to Blob'
+      condition: and(succeeded(), eq(variables['NPM_PUBLISH_FLAG'], 'true'))
+      inputs:
+        targetType: 'inline'
+        script: |
+          export AZCOPY_SPA_CLIENT_SECRET=$(AZURESDKPARTNERDROPS-SERVICE-PRINCIPAL-KEY)
+          azcopy login --service-principal --application-id $(APPID)
+          azcopy copy "$(Build.ArtifactStagingDirectory)/Package/*"  "https://azuresdkpartnerdrops.blob.core.windows.net/drops/azureiotedge/npm/$(PACKAGEVERSION)"
```

Comment:
>* [](http://example.com/codeflow?start=65&length=1)

we only want to publish the tgz file, not the manifests

Reply:
we have copy task in line 39 to isolate the tgz file to Package

Reply:
Instead of maintaining a separate folder for the tgz file, is it possible to simply reference is int he assets section as *.tgz? This way we won't need to have a separate copyfile task on line 39